### PR TITLE
🔨🎉📙 Refine annotation workflows

### PR DESCRIPTION
**Pull request by migbro**
# Pull Request Template

## Description

Changed output capture from file merge to directory output for the annotator split workflows.
File merge not needed by Jeremy for variant warehouse load.
Documentation on how input files for precalculation were generated.

Contributes to closing https://app.zenhub.com/workspaces/d3b-bfx-5cdd720eed0dce4209e0b8a5/issues/d3b-center/bixu-tracker/622 and closes https://app.zenhub.com/workspaces/d3b-bfx-5cdd720eed0dce4209e0b8a5/issues/d3b-center/bixu-tracker/589

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Successfully run on cavatica
- [ ] Passes cwltool

**Test Configuration**:
* Environment: Cavatica
* Test files: Tasks: https://cavatica.sbgenomics.com/u/d3b-bixu/dev-wgsa/tasks/#q?search=DBSNP%20INDEL%20V153&page=1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: tools/output_to_dir.cwl [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                    cmd += "echo \"cp " + inputs.input_array[i].path + " " + result_dir + "/" + i.toString()
+                    + "_" + inputs.input_array[i].basename + "\" >> cmd_list.txt;";
+                    if (inputs.input_array[i].secondaryFiles){
+                      cmd += "echo \"cp " + inputs.input_array[i].secondaryFiles[0].path + " " + result_dir + "/" + i.toString()
+                    + "_" + inputs.input_array[i].secondaryFiles[0].basename + "\" >> cmd_list.txt;";
+                    }
+            }
+          }
+          return cmd;
+      }
```
Comment (dmiller15) [inline tools/output_to_dir.cwl:65]: 👀

File: tools/output_to_dir.cwl [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          cmd += "mkdir " + result_dir + ";";
+          var out_dirs = []
+          var protocol_dict = new Object();
+          if(inputs.input_scatter != null){
+            for (i = 0; i < inputs.protocol_name.length; i++) {
+                out_dirs.push(result_dir + "/" + inputs.protocol_name[i]);
+                cmd += "mkdir " + out_dirs[i] + ";";
+                protocol_dict[inputs.protocol_name[i]] = 0;
+            }
+              i = 0;
```
Comment (dmiller15) [inline tools/output_to_dir.cwl:34]: this indentation looks wrong

File: tools/output_to_dir.cwl [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      ${
+          var cmd = "";
+          var result_dir = inputs.output_basename + "_" + inputs.tool_name;
+          var i;
+          cmd += "mkdir " + result_dir + ";";
+          var out_dirs = []
+          var protocol_dict = new Object();
+          if(inputs.input_scatter != null){
+            for (i = 0; i < inputs.protocol_name.length; i++) {
+                out_dirs.push(result_dir + "/" + inputs.protocol_name[i]);
```
Comment (dmiller15) [inline tools/output_to_dir.cwl:30]: inconsistent indentation

File: tools/output_to_dir.cwl (hunk 1 above)
Comment (dmiller15) [reply tools/output_to_dir.cwl:65]: I think i get it but this requires documentation.

**Comment by migbro** [comment]
Tested tool update with commenting here: https://cavatica.sbgenomics.com/u/d3b-bixu/dev-wgsa/tasks/#q?search=merge-outputs-to-dir&page=1

**Pull request merged**